libs/commons/Common.mli

```diff
@@ -438,6 +438,19 @@ val partition_result :
 
 val memoized : ?use_cache:bool -> ('a, 'b) Hashtbl.t -> 'a -> (unit -> 'b) -> 'b
 
+(*****************************************************************************)
+(* Composition/Control *)
+(*****************************************************************************)
+
+val protect : finally:(unit -> unit) -> (unit -> 'a) -> 'a
+(** Same as 'Fun.protect' but we block SIGALRM while executing `finally()`, this
+ * prevents that a timeout set via 'Time_limit' accidentally raises 'Timeout'
+ * in the middle of 'finally's code, which would crash Semgrep. After 'finally'
```

Comment:
Why it would crash Semgrep?

Reply:
It's a programming error raising `Timeout` inside a `finally` and it results in a `Finally_raised` exception that we don't handle. We should probably handle it though, so that it doesn't crash at least.

Reply:
Hmmm, I'm not sure why it is not caught in `iter_targets_and_get_matches_and_exn_to_errors` to be honest.

Reply:
Ah yeah, sorry, it's because I got these crashes from Semgrep Pro, where we are not catching all the exceptions.